### ISSUE TITLE
provider/fastly: fastly_service_v1.default_ttl doesn't update when it's 0 

### ISSUE THREAD

**Issue by premist** [hashicorp/terraform#12910]
### Terraform Version

Terraform 0.9.1

### Affected Resources

- fastly_service_v1

### Terraform Configuration

```hcl
resource "fastly_service_v1" "frontend" {
  name = "shakr-xxx-frontend"
  default_ttl = 0

  ...
}
```

### Outputs

On terraform plan:

```shell
fastly_service_v1.frontend: Refreshing state... (ID: xxx)

~ fastly_service_v1.frontend
    default_ttl: "60" => "0"
```

On terraform apply:

```shell
fastly_service_v1.frontend: Modifying...
  default_ttl: "60" => "0"

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

### Expected Behavior
It should update `default_ttl` attribute to 0.

### Actual Behavior
It didn't update `default_ttl` attribute to 0. Consecutive `terraform plan` command shows `fastly_service_v1.frontend` as a resource that should be updated with the same change (changing `default_ttl` to 0 from 60), and TTL is not changed either on Fastly dashboard.

In comparison, providing number other than 0 (like 3600) works fine. Updating this number on Fastly dashboard also works fine.

### Steps to Reproduce
1. Create Terraform configuration with default_ttl with a number larger than 0 (e.g. 60)
1. Create a resource by using `terraform apply`
1. Modify Terraform configuration with default_ttl with a number 0
1. Update a resource by using `terraform apply`

**Comment by aerostitch** [comment]
See my reply in #13648. Sorry I thought it would have been an easier fix but that's mainly an issue in the go-fastly api.

**Comment by sethvargo** [comment]
Merged

**Comment by aerostitch** [comment]
Thanks! :)
I'll wait to see what we do on https://github.com/sethvargo/go-fastly/pull/40 before updating the PR with he latest version of go-fastly as it might also affect the fetching of the latest version used in here.

**Comment by aerostitch** [comment]
All the upstream changes have been merged and I pushed the upgrade to the Terraform PR.

**Comment by catsby** [comment]
I merged #13648 which fixes this, thanks!
